### ISSUE TITLE
API: define response type in config and via parameter

### ISSUE THREAD

**Issue by stefanorosanelli** [bedita/bedita#680]
Default response type is currently JSON - which is fine
We should be able to define response type via config `apiResponseType` property or through `respType=XML` URL parameter (names may obviously change)

In order of priority:
1. URL parameter
2. configuration
3. current setup (default or `ResponseHandler` definition in controller)

**Comment by batopa** [comment]
To return a good xml we should change the response used for json because it's not well formed.
For example `/api/objects/1/children` returns something like

``` xml
<data>
    <objects></objects>
    <objects></objects>
    <objects></objects>
    ...
    <related></related>
    <related></related>
</data>
```

it would be better something like

``` xml
<data>
    <objects>
        <object></object>
        <object></object>
        ...
    </objects>

    <related>
        <object></object>
        <object></object>
        ...
    </related>
</data>
```

This should be applied for all arrays as tags, categories, etc...

We can apply some other transformations in **xml** case or avoiding to serve xml response.

Do we really need xml response? Honestly I would prefer to stay focused on **json** response. By now JSON response is the most common format for API and JSON parser exists for every language so I don't see any advantage to support also xml at least at this initial stage.

@stefanorosanelli @qwerg @edoardocavazza @xho what do you think?

**Comment by edoardocavazza** [comment]
I absolutely agree with you

**Comment by stefanorosanelli** [comment]
Let's stay focused on **JSON** as primary response type, let's keep (partial) **XML** support since it's still used. Customers still ask for it, so we need it.
We can:
- recommend use of **JSON** anyway
- warn that **XML** support is not optimal (like in previous example) - use it at your own risk! :)

**Comment by stefanorosanelli** [comment]
resolved in #836
